crates/cxx-qt-build/src/lib.rs

```diff
@@ -715,26 +767,104 @@ impl CxxQtBuilder {
         // https://github.com/rust-lang/rust/issues/108081
         // https://github.com/KDAB/cxx-qt/pull/598
         if qtbuild.version().major == 5 {
-            let std_types_contents = include_str!("std_types_qt5.cpp");
-            let std_types_path = format!(
-                "{out_dir}/std_types_qt5.cpp",
-                out_dir = env::var("OUT_DIR").unwrap()
-            );
-            let mut source =
-                File::create(&std_types_path).expect("Could not create std_types source");
-            write!(source, "{std_types_contents}").expect("Could not write std_types source");
-            cc_builder_whole_archive.file(&std_types_path);
-            cc_builder_whole_archive_files_added = true;
+            self.initializers
+                .push(include_str!("std_types_qt5.cpp").to_owned());
         }
+    }
 
-        if cc_builder_whole_archive_files_added {
-            cc_builder_whole_archive.compile("qt-static-initializers");
+    fn build_initializers(&mut self, init_builder: &cc::Build) {
+        let initializers_path = out_dir().join("cxxqt_initializers.cpp");
+        std::fs::write(&initializers_path, self.initializers.join("\n"))
+            .expect("Could not write cxx_qt_initializers.cpp");
+        Self::build_object_file(
+            init_builder,
+            initializers_path,
+            Some(("", "initializers.o")),
+        );
+    }
+
+    fn build_qrc_files(&mut self, init_builder: &cc::Build, qtbuild: &mut qt_build_utils::QtBuild) {
+        for qrc_file in &self.qrc_files {
+            let obj_file_dir = format!("qrc/{}", crate_name());
+            let obj_file_name = format!(
+                "{}.o",
+                qrc_file.file_name().unwrap_or_default().to_string_lossy()
+            );
+            // TODO: Is it correct to use an obj file here, or should this just link normally?
```

Comment:
looking around qrc files are normally converted to a `.cpp` file and then compiled to a `.cpp.o` file in Qt CMake, then i assume that is linked in.

Reply:
then there is also a `_init` cpp and cpp.o as well for the initialiser

Reply:
Hm, qt-build-utils currently only dumps out the one rcc file, but splitting this would be a way to go in the future I guess :thinking: